roles/base/tasks/main.yml

```diff
@@ -1,72 +1,222 @@
+    state: present
+    create: yes
+    owner: root
+    group: root
+    mode: 0644
+  when: ansible_os_family == "RedHat"
+
+- name: Add Postgres APT GPG key (Ubuntu)
+  apt_key: 
+    state: present
+    url: "https://www.postgresql.org/media/keys/ACCC4CF8.asc"
+  when: ansible_os_family == "Debian"
+
+- name: Install Postgres repo (Ubuntu)
+  apt_repository:
+    repo: "deb http://apt.postgresql.org/pub/repos/apt/ {{ansible_distribution_release}}-pgdg main"
+    state: present
+  when: ansible_os_family == "Debian"
+
+- name: Install certbot
+  package:
+    name: "{{certbot_package}}"
+    update_cache: yes
+
+- name: Generate certbot cert (PRD)
+  command: "{{certbot_package}} certonly -n --keep --standalone -d {{ansible_fqdn}}  --agree-tos --email {{certbot_username}}"
+  args:
+    creates: "/etc/letsencrypt/live/{{ansible_fqdn}}"
+  when: oval_environment == "PRD"
+
+- name: Generate certbot cert (STAGING)
+  command: "{{certbot_package}} certonly --staging -n --keep --standalone -d {{ansible_fqdn}}  --agree-tos --email {{certbot_username}}"
+  args:
+    creates: "/etc/letsencrypt/live/{{ansible_fqdn}}"
+  when: oval_environment != "PRD"
+
+- name: Create certbot hooks dir
+  file: 
+    path: /etc/letsencrypt/hooks
+    state: directory
+    mode: 0755
+    owner: root
+    group: root
+
+- name: Install certbot cronjob (PRD)
+  cron:
+    name: "certbot cert renewal"
+    special_time: daily
+    cron_file: certbot
+    user: root
+    state: present
+    job: "/usr/bin/{{certbot_package}} renew -n --keep --standalone --pre-hook /etc/letsencrypt/hooks/pre.sh --post-hook /etc/letsencrypt/hooks/post.sh"
+  when: oval_environment == "PRD"
+
+- name: Install certbot cronjob (STAGING)
+  cron:
+    name: "certbot cert renewal"
+    special_time: daily
+    cron_file: certbot
+    user: root
+    state: present
+    job: "/usr/bin/{{certbot_package}} renew -n --keep --standalone --pre-hook /etc/letsencrypt/hooks/pre.sh --post-hook /etc/letsencrypt/hooks/post.sh"
+  when: oval_environment != "PRD"
 
 - name: Install sysstat
-  yum: name=sysstat update_cache=yes
+  package: 
+    name: sysstat
+    update_cache: yes
 
 - name: Enable sysstat
-  service: name=sysstat state=started enabled=yes
-
-- command: grep "^/swapfile" /proc/swaps
-  register: swapfile_result 
+  service:
+    name: sysstat
+    state: started
+    enabled: yes
+
+- name: Check for an active swapfile
+  command: grep "^/swapfile" /proc/swaps
+  register: swapfile_result
   ignore_errors: true
 
 - name: Create swap file
   command: dd if=/dev/zero of=/swapfile bs=1M count=2048
   when: swapfile_result.rc != 0
 
 - name: Format swap file
-  command: mkswap /swapfile 
+  command: mkswap /swapfile
   when: swapfile_result.rc != 0
-  
+
 - name: Change perms on swap file
-  file: path=/swapfile owner=root group=root mode=0600
+  file: 
+    path: /swapfile
+    owner: root
+    group: root
+    mode: 0600
 
 - name: Swapon swap file
   command: swapon /swapfile
```

Comment:
No, but we could if you want to. I know how strongly you feel about encrypting all of the things.